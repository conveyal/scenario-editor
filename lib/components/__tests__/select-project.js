/* global describe, it, expect, jest */

describe('Component > SelectProject', () => {
  const renderer = require('react-test-renderer')
  const React = require('react')
  const SelectProject = require('../select-project')

  it('renders correctly', () => {
    const createFn = jest.fn()
    const mockProjects = [{ id: 1, name: 'P1' }, { id: 2, name: 'P2' }]
    const pushFn = jest.fn()
    const loadAllProjects = jest.fn()
    const tree = renderer.create(
      <SelectProject
        create={createFn}
        loadAllProjects={loadAllProjects}
        projects={mockProjects}
        push={pushFn}
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
    expect(createFn).not.toBeCalled()
    expect(pushFn).not.toBeCalled()
    expect(loadAllProjects).toBeCalled()
  })
})
