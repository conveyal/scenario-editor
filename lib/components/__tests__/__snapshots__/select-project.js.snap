// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > SelectProject 1`] = `
<SelectProject
  bundles={
    Array [
      Object {
        "_id": "1",
        "accessGroup": "accessGroupName",
        "centerLat": 38.90124,
        "centerLon": -77.015615,
        "createdAt": "2017-11-08T12:30:50.133Z",
        "createdBy": "person@gmail.com",
        "errorCode": "",
        "feeds": Array [],
        "name": "Mock Bundle",
        "nonce": "12345",
        "regionId": "1",
        "serviceEnd": "2018-09-28",
        "serviceStart": "2018-09-27",
        "status": "DONE",
        "totalFeeds": 0,
        "updatedAt": "2017-11-08T12:31:19.602Z",
        "updatedBy": "person@gmail.com",
      },
    ]
  }
  projects={
    Array [
      Object {
        "_id": "1",
        "accessGroup": "accessGroupName",
        "bundleId": "1",
        "createdAt": "2017-11-08T12:30:50.133Z",
        "createdBy": "person@gmail.com",
        "name": "Mock Project",
        "nonce": "12345",
        "regionId": "1",
        "updatedAt": "2017-11-08T12:31:19.602Z",
        "updatedBy": "person@gmail.com",
        "variants": Array [
          "Default",
        ],
      },
    ]
  }
  region={
    Object {
      "_id": "1",
      "accessGroup": "accessGroupName",
      "bounds": Object {
        "east": -76,
        "north": 39,
        "south": 38,
        "west": -77,
      },
      "createdAt": "2017-11-08T12:30:50.133Z",
      "createdBy": "person@gmail.com",
      "description": "Region description",
      "name": "Mock Region",
      "nonce": "12345",
      "opportunityDatasets": Array [
        Object {
          "_id": "Workers_total",
          "name": "Workers total",
          "sourceName": "lodes-data-2014",
        },
      ],
      "statusCode": "DONE",
      "updatedAt": "2017-11-08T12:31:19.602Z",
      "updatedBy": "person@gmail.com",
    }
  }
>
  <InnerDock>
    <SimpleBar
      className="InnerDock "
      style={
        Object {
          "height": 768,
        }
      }
    >
      <div
        className="InnerDock "
        data-simplebar={true}
        style={
          Object {
            "height": 768,
          }
        }
      >
        <div
          className="simplebar-wrapper"
        >
          <div
            className="simplebar-height-auto-observer-wrapper"
          >
            <div
              className="simplebar-height-auto-observer"
            />
          </div>
          <div
            className="simplebar-mask"
          >
            <div
              className="simplebar-offset"
            >
              <div
                className="simplebar-content-wrapper"
              >
                <div
                  className="simplebar-content"
                >
                  <Stack
                    px={2}
                    py={4}
                    spacing={4}
                  >
                    <Styled(div)
                      __css={
                        Object {
                          "& > *:not(style) ~ *:not(style)": Object {
                            "marginStart": 0,
                            "marginTop": 4,
                          },
                        }
                      }
                      className="chakra-stack"
                      display="flex"
                      flexDirection="column"
                      px={2}
                      py={4}
                    >
                      <div
                        className="chakra-stack css-ct7syb"
                      >
                        <Flex
                          align="center"
                          key=".0"
                          pl={2}
                        >
                          <Styled(div)
                            __css={
                              Object {
                                "alignItems": "center",
                                "display": "flex",
                                "flexBasis": undefined,
                                "flexDirection": undefined,
                                "flexGrow": undefined,
                                "flexShrink": undefined,
                                "flexWrap": undefined,
                                "justifyContent": undefined,
                              }
                            }
                            pl={2}
                          >
                            <div
                              className="css-1ruoioj"
                            >
                              <Heading
                                flex="1"
                                size="md"
                              >
                                <Styled(h2)
                                  __css={
                                    Object {
                                      "fontFamily": "heading",
                                      "fontSize": "xl",
                                      "fontWeight": "bold",
                                      "lineHeight": 1.2,
                                    }
                                  }
                                  className="chakra-heading"
                                  flex="1"
                                >
                                  <h2
                                    className="chakra-heading css-o8drfs"
                                  >
                                    Mock Region
                                  </h2>
                                </Styled(h2)>
                              </Heading>
                              <ForwardRef
                                label="Edit region settings"
                                onClick={[Function]}
                              >
                                <Box>
                                  <div
                                    className="css-0"
                                  >
                                    <ForwardRef
                                      label="Edit region settings"
                                      placement="auto"
                                    >
                                      <Tooltip
                                        aria-label="Tooltip: Edit region settings"
                                        fontSize="md"
                                        label="Edit region settings"
                                        placement="auto"
                                        zIndex={1500}
                                      >
                                        <Button
                                          aria-label="Edit region settings"
                                          colorScheme="blue"
                                          isActive={false}
                                          isDisabled={false}
                                          onBlur={[Function]}
                                          onClick={[Function]}
                                          onFocus={[Function]}
                                          onMouseDown={[Function]}
                                          onMouseEnter={[Function]}
                                          size="sm"
                                          variant="ghost"
                                        >
                                          <Styled(button)
                                            __css={
                                              Object {
                                                "_active": Object {
                                                  "bg": "blue.100",
                                                },
                                                "_disabled": Object {
                                                  "boxShadow": "none",
                                                  "cursor": "not-allowed",
                                                  "opacity": 0.4,
                                                },
                                                "_focus": Object {
                                                  "boxShadow": "outline",
                                                },
                                                "_hover": Object {
                                                  "_disabled": Object {
                                                    "bg": "initial",
                                                  },
                                                  "bg": "blue.50",
                                                },
                                                "alignItems": "center",
                                                "appearance": "none",
                                                "bg": "transparent",
                                                "borderRadius": "md",
                                                "color": "blue.600",
                                                "display": "inline-flex",
                                                "fontSize": "sm",
                                                "fontWeight": "semibold",
                                                "h": 8,
                                                "justifyContent": "center",
                                                "lineHeight": "1.2",
                                                "minW": 8,
                                                "outline": "none",
                                                "position": "relative",
                                                "px": 3,
                                                "transition": "all 250ms",
                                                "userSelect": "none",
                                                "verticalAlign": "middle",
                                                "whiteSpace": "nowrap",
                                                "width": "auto",
                                              }
                                            }
                                            aria-label="Edit region settings"
                                            className="chakra-button"
                                            onBlur={[Function]}
                                            onClick={[Function]}
                                            onFocus={[Function]}
                                            onMouseDown={[Function]}
                                            onMouseEnter={[Function]}
                                            type="button"
                                          >
                                            <button
                                              aria-label="Edit region settings"
                                              className="chakra-button css-1o9h2ju"
                                              onBlur={[Function]}
                                              onClick={[Function]}
                                              onFocus={[Function]}
                                              onMouseDown={[Function]}
                                              onMouseEnter={[Function]}
                                              type="button"
                                            >
                                              <FaCog>
                                                <IconBase
                                                  attr={
                                                    Object {
                                                      "viewBox": "0 0 512 512",
                                                    }
                                                  }
                                                >
                                                  <svg
                                                    fill="currentColor"
                                                    height="1em"
                                                    stroke="currentColor"
                                                    strokeWidth="0"
                                                    style={
                                                      Object {
                                                        "color": undefined,
                                                      }
                                                    }
                                                    viewBox="0 0 512 512"
                                                    width="1em"
                                                    xmlns="http://www.w3.org/2000/svg"
                                                  >
                                                    <path
                                                      d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                                                      key="0"
                                                    />
                                                  </svg>
                                                </IconBase>
                                              </FaCog>
                                            </button>
                                          </Styled(button)>
                                        </Button>
                                        <AnimatePresence />
                                      </Tooltip>
                                    </ForwardRef>
                                  </div>
                                </Box>
                              </ForwardRef>
                            </div>
                          </Styled(div)>
                        </Flex>
                        <Stack
                          key=".1"
                          px={2}
                        >
                          <Styled(div)
                            __css={
                              Object {
                                "& > *:not(style) ~ *:not(style)": Object {
                                  "marginStart": 0,
                                  "marginTop": "0.5rem",
                                },
                              }
                            }
                            className="chakra-stack"
                            display="flex"
                            flexDirection="column"
                            px={2}
                          >
                            <div
                              className="chakra-stack css-i652e9"
                            >
                              <Button
                                colorScheme="green"
                                isFullWidth={true}
                                key=".0"
                                onClick={[Function]}
                              >
                                <Styled(button)
                                  __css={
                                    Object {
                                      "_active": Object {
                                        "bg": "green.700",
                                      },
                                      "_disabled": Object {
                                        "boxShadow": "none",
                                        "cursor": "not-allowed",
                                        "opacity": 0.4,
                                      },
                                      "_focus": Object {
                                        "boxShadow": "outline",
                                      },
                                      "_hover": Object {
                                        "_disabled": Object {
                                          "bg": "green.500",
                                        },
                                        "bg": "green.600",
                                      },
                                      "alignItems": "center",
                                      "appearance": "none",
                                      "bg": "green.500",
                                      "borderRadius": "md",
                                      "color": "white",
                                      "display": "inline-flex",
                                      "fontSize": "md",
                                      "fontWeight": "semibold",
                                      "h": 10,
                                      "justifyContent": "center",
                                      "lineHeight": "1.2",
                                      "minW": 10,
                                      "outline": "none",
                                      "position": "relative",
                                      "px": 4,
                                      "transition": "all 250ms",
                                      "userSelect": "none",
                                      "verticalAlign": "middle",
                                      "whiteSpace": "nowrap",
                                      "width": "100%",
                                    }
                                  }
                                  className="chakra-button"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <button
                                    className="chakra-button css-1uiu0j2"
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    Create new Project
                                  </button>
                                </Styled(button)>
                              </Button>
                            </div>
                          </Styled(div)>
                        </Stack>
                        <Box
                          key=".2"
                          textAlign="center"
                        >
                          <div
                            className="css-xi606m"
                          >
                            or go to an existing project
                          </div>
                        </Box>
                        <Stack
                          key=".3"
                          px={2}
                          spacing={0}
                        >
                          <Styled(div)
                            __css={
                              Object {
                                "& > *:not(style) ~ *:not(style)": Object {
                                  "marginStart": 0,
                                  "marginTop": 0,
                                },
                              }
                            }
                            className="chakra-stack"
                            display="flex"
                            flexDirection="column"
                            px={2}
                          >
                            <div
                              className="chakra-stack css-obiul0"
                            >
                              <Project
                                key=".$1"
                                project={
                                  Object {
                                    "_id": "1",
                                    "accessGroup": "accessGroupName",
                                    "bundleId": "1",
                                    "createdAt": "2017-11-08T12:30:50.133Z",
                                    "createdBy": "person@gmail.com",
                                    "name": "Mock Project",
                                    "nonce": "12345",
                                    "regionId": "1",
                                    "updatedAt": "2017-11-08T12:31:19.602Z",
                                    "updatedBy": "person@gmail.com",
                                    "variants": Array [
                                      "Default",
                                    ],
                                  }
                                }
                              >
                                <ListGroupItem
                                  leftIcon={<FaCube />}
                                  onClick={[Function]}
                                >
                                  <Button
                                    _notFirst={
                                      Object {
                                        "borderTopLeftRadius": 0,
                                        "borderTopRightRadius": 0,
                                      }
                                    }
                                    _notLast={
                                      Object {
                                        "borderBottom": 0,
                                        "borderBottomLeftRadius": 0,
                                        "borderBottomRightRadius": 0,
                                      }
                                    }
                                    borderColor="#E2E8F0"
                                    borderRadius="4px"
                                    colorScheme="blue"
                                    height="unset"
                                    isFullWidth={true}
                                    justifyContent="left"
                                    leftIcon={<FaCube />}
                                    onClick={[Function]}
                                    py={4}
                                    textAlign="left"
                                    variant="outline"
                                    whiteSpace="break-spaces"
                                  >
                                    <Styled(button)
                                      __css={
                                        Object {
                                          "_active": Object {
                                            "bg": "blue.100",
                                          },
                                          "_disabled": Object {
                                            "boxShadow": "none",
                                            "cursor": "not-allowed",
                                            "opacity": 0.4,
                                          },
                                          "_focus": Object {
                                            "boxShadow": "outline",
                                          },
                                          "_hover": Object {
                                            "_disabled": Object {
                                              "bg": "initial",
                                            },
                                            "bg": "blue.50",
                                          },
                                          "alignItems": "center",
                                          "appearance": "none",
                                          "bg": "transparent",
                                          "border": "1px solid",
                                          "borderColor": "currentColor",
                                          "borderRadius": "md",
                                          "color": "blue.600",
                                          "display": "inline-flex",
                                          "fontSize": "md",
                                          "fontWeight": "semibold",
                                          "h": 10,
                                          "justifyContent": "center",
                                          "lineHeight": "1.2",
                                          "minW": 10,
                                          "outline": "none",
                                          "position": "relative",
                                          "px": 4,
                                          "transition": "all 250ms",
                                          "userSelect": "none",
                                          "verticalAlign": "middle",
                                          "whiteSpace": "nowrap",
                                          "width": "100%",
                                        }
                                      }
                                      _notFirst={
                                        Object {
                                          "borderTopLeftRadius": 0,
                                          "borderTopRightRadius": 0,
                                        }
                                      }
                                      _notLast={
                                        Object {
                                          "borderBottom": 0,
                                          "borderBottomLeftRadius": 0,
                                          "borderBottomRightRadius": 0,
                                        }
                                      }
                                      borderColor="#E2E8F0"
                                      borderRadius="4px"
                                      className="chakra-button"
                                      height="unset"
                                      justifyContent="left"
                                      onClick={[Function]}
                                      py={4}
                                      textAlign="left"
                                      type="button"
                                      whiteSpace="break-spaces"
                                    >
                                      <button
                                        className="chakra-button css-3aw1nz"
                                        onClick={[Function]}
                                        type="button"
                                      >
                                        <ButtonIcon
                                          marginEnd="0.5rem"
                                        >
                                          <Styled(span)
                                            className="chakra-button__icon"
                                            marginEnd="0.5rem"
                                          >
                                            <span
                                              className="chakra-button__icon css-ktvjz1"
                                            >
                                              <FaCube
                                                aria-hidden={true}
                                                focusable={false}
                                              >
                                                <IconBase
                                                  aria-hidden={true}
                                                  attr={
                                                    Object {
                                                      "viewBox": "0 0 512 512",
                                                    }
                                                  }
                                                  focusable={false}
                                                >
                                                  <svg
                                                    aria-hidden={true}
                                                    fill="currentColor"
                                                    focusable={false}
                                                    height="1em"
                                                    stroke="currentColor"
                                                    strokeWidth="0"
                                                    style={
                                                      Object {
                                                        "color": undefined,
                                                      }
                                                    }
                                                    viewBox="0 0 512 512"
                                                    width="1em"
                                                    xmlns="http://www.w3.org/2000/svg"
                                                  >
                                                    <path
                                                      d="M239.1 6.3l-208 78c-18.7 7-31.1 25-31.1 45v225.1c0 18.2 10.3 34.8 26.5 42.9l208 104c13.5 6.8 29.4 6.8 42.9 0l208-104c16.3-8.1 26.5-24.8 26.5-42.9V129.3c0-20-12.4-37.9-31.1-44.9l-208-78C262 2.2 250 2.2 239.1 6.3zM256 68.4l192 72v1.1l-192 78-192-78v-1.1l192-72zm32 356V275.5l160-65v133.9l-160 80z"
                                                      key="0"
                                                    />
                                                  </svg>
                                                </IconBase>
                                              </FaCube>
                                            </span>
                                          </Styled(span)>
                                        </ButtonIcon>
                                        Mock Project
                                      </button>
                                    </Styled(button)>
                                  </Button>
                                </ListGroupItem>
                              </Project>
                            </div>
                          </Styled(div)>
                        </Stack>
                      </div>
                    </Styled(div)>
                  </Stack>
                </div>
              </div>
            </div>
          </div>
          <div
            className="simplebar-placeholder"
          />
        </div>
        <div
          className="simplebar-track simplebar-horizontal"
        >
          <div
            className="simplebar-scrollbar"
          />
        </div>
        <div
          className="simplebar-track simplebar-vertical"
        >
          <div
            className="simplebar-scrollbar"
          />
        </div>
      </div>
    </SimpleBar>
  </InnerDock>
</SelectProject>
`;
